ethereum/simplestack/simple_stack.go

```diff
@@ -0,0 +1,167 @@
+package simplestack
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/0xProject/0x-mesh/ethereum/miniheader"
+)
+
+type updateType int
+
+const (
+	Pop updateType = iota
+	Push
+)
+
+// Update represents one update to the stack, either a pop or push of a miniHeader.
+type Update struct {
+	Type       updateType
+	MiniHeader *miniheader.MiniHeader
+}
+
+// SimpleStack is a simple in-memory stack used in tests
+type SimpleStack struct {
+	limit              int
+	miniHeaders        []*miniheader.MiniHeader
+	updates            []*Update
+	mu                 sync.Mutex
+	latestCheckpointID int
+}
+
+// NewSimpleStack instantiates a new SimpleStack
+func NewSimpleStack(retentionLimit int, miniHeaders []*miniheader.MiniHeader) *SimpleStack {
```

Comment:
This should just be called `New` since it's already namespaced in the `simplestack` package.